Csv/CsvReader.cs

```diff
@@ -237,15 +317,14 @@ private string[] Line
                 {
                     if (parsedLine == null)
                     {
-                        lock (headerLookup)
-                        {
-                            if (parsedLine == null)
-                                parsedLine = SplitLine(Raw, options);
+                        var raw = RawSplitLine;
 
-                            if (options.ValidateColumnCount && parsedLine.Length != Headers.Length)
-                                throw new InvalidOperationException($"Expected {Headers.Length}, got {parsedLine.Length} columns.");
-                        }
+                        if (options.ValidateColumnCount && raw.Length != Headers.Length)
```

Comment:
I wanted to make sure this validation happened after the split but before the trim, so that an invalid ReadLine (with unterminated quote-enclosed value) could be detected and recreated with the entire record value.